### DISCUSSION TITLE
[BUG]: cannot put static files like javascript or css

### DISCUSSION THREAD

**Discussion by BanDroid** [benallfree/pocketpages#19] [Q&A] [answered]
any idea how to put js or css file in pages? i cant use any js or css when linked to my ejs page, strict MIME type enabled.

**Comment by benallfree** [comment]
Hello, can you please explain where you are placing the files? Show me your directory structure.

**Reply by BanDroid** [reply]
![structure](https://github.com/user-attachments/assets/e0d8d395-747e-40c9-9274-8c21689036b1)

my code to load the script and css is like this

```html
<!DOCTYPE html>
<html lang="id">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Portfolio Material 3</title>
    <!-- alpinejs -->
    <script src="/static/libs/alpine.min.js" defer></script>

    <!-- mdui -->
    <link rel="stylesheet" href="/static/libs/mdui.css" />
    <script src="/static/libs/mdui.global.js"></script>
  </head>
  <body>
    <h1>hello</h1>
    <%- slot %>
  </body>
</html>
```

**Reply by BanDroid** [reply]
it gives me this error in console

![console-error](https://github.com/user-attachments/assets/d9f0744d-ed46-4030-ad61-c2a9950103eb)

**Reply by BanDroid** [reply]
using attribute `type="text/javascript"` or `type="application/javascript"` for js and `type="text/css"` for css did not work too.

**Reply by benallfree** [reply]
Let me research this.

**Reply by BanDroid** [reply]
the bug is confirmed right? should you put this as issue then?

**Reply by benallfree** [reply]
I cannot reproduce this bug.

```js
// pb_hooks/pages/test.js
console.log('hello world')
```

```ejs
// pb_hooks/pages/index.ejs

<script src="/test.js"></script>
```

```
pb_hooks/
   pages/
      index.ejs
      test.js
```

<img width="840" alt="image" src="https://github.com/user-attachments/assets/e2842a51-db1c-4cc6-b10f-35149fbc075a">

**Reply by BanDroid** [reply]
is it browser related? i use chrome, and pocketbase version 0.22.22

**Reply by BanDroid** [reply]
oh yeah, i use the script inside layout, not index.ejs (page)

**Comment by BanDroid** [comment, marked as answer]
i need to serve static files in `pb_public` instead.
altough, i dont know how yours working inside `pb_hooks/pages`.
welp, i have suspicion on the file size limit on serving static files in `pb_hooks/pages`, somehow i put my alpinejs and mdui script into `pb_public`, then i leave 1 script `test.js` which is just logging "hello world", the "hello world" logged to the console
but when my alpinejs and mdui script inside `pb_hooks/pages`, all of my script and css would fail to recognize the mimetype.
even `test.js` would fail too when they're together inside `pb_hooks/pages`.

**Reply by BanDroid** [reply]
so maybe we should leave `pb_hooks/pages` for pages only, and serve any script or css in `pb_public` instead.
i will mark this as answer.

**Reply by benallfree** [reply]
That's not the correct approach, please send me a repo link so I can see.

**Comment by c-nv-s** [comment]
```
pb_hooks/
   pages/
      (site)/
      xapi/
         mypage/
            _private/
               test.js
            mypage.ejs
```

```
// pb_hooks/pages/xapi/mypage/_private/test.js
console.log('hello from test.js')
```

```
// pb_hooks/pages/xapi/mypage/mypage.ejs
<script> console.log('hello from mypage.ejs') </script>
<script src="/xapi/mypage/_private/test.js"></script>
```

I am seeing that I get no scripts loaded in the browser developer tools nor the ``console.log()`` messages appearing there.

**Reply by benallfree** [reply]
`_private` files are not routable.

**Comment by benallfree** [comment]
I'm closing this issue because it's beginning to wander off its original question. Please feel free to post fresh topics.
